mirrord-layer/src/lib.rs

```diff
@@ -450,3 +466,46 @@ unsafe extern "C" fn close_detour(fd: c_int) -> c_int {
         FN_CLOSE(fd)
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use rstest::*;
+
+    use super::*;
+
+    #[rstest]
```

Comment:
if you're leveraging rstest, most of the values can be parameterized/cases and then have test for not load and case for load..?